### PR TITLE
setup_admin

### PR DESCRIPTION
**Pull request by hongthamnguyen2k**
#4

- Setup gem https://github.com/heartcombo/devise 
- Create Admin tables
- Update file seed.rb (use faker, factory_bot create sample admin) ==> (Khoa & anh NAnh)
- Design login form, use gem devise allow admin login to admin page and logout ==> (Thắm)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,13 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
```
[the remark this answers is not in the document]
Comment (hongthamnguyen2k) [reply app/controllers/application_controller.rb:3]: rubocop said: 1 offense detected: Extra empty line detected at class body beginning.

**Review by khoa250200 — APPROVED** [review]

**Review by anhpn0712 — APPROVED** [review]

File: app/assets/javascript/admin/js/main.js
Comment (KhoaNDAHopee) [inline app/assets/javascript/admin/js/main.js:1]: Check indent

File: app/assets/stylesheets/admin/application.scss [hunk 2]
```diff
@@ -2,3 +2,8 @@
 //= require "../style/css/vendor/fullcalendar.min"
 //= require "../style/css/icons.min"
 //= require "../style/css/app.min"
+//= require "../style/css/style"
+
+.img {
+  background-image: url("assets/style/images/bg_1.jpg")
```
Comment (KhoaNDAHopee) [inline app/assets/stylesheets/admin/application.scss]: Don't use background-image for all class .img on all page (should be for login page)

File: app/models/admin.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Admin < ApplicationRecord
```
Comment (KhoaNDAHopee) [inline app/models/admin.rb:3]: Should be validate for some fields

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html>
+<html lang="en">
+   <head>
```
Comment (KhoaNDAHopee) [inline app/views/devise/sessions/new.html.erb]: Header should be in layout, not here.

File: app/views/shared/partials/_navbar.html.erb [hunk 5]
```diff
@@ -136,9 +136,8 @@
         <span>Lock Screen</span>
         </a>
         <!-- item-->
-        <a href="javascript:void(0);" class="dropdown-item notify-item">
-        <i class="mdi mdi-logout me-1"></i>
-        <span>Logout</span>
+        <a href="javascript:void(0);">
+        <span><%= link_to "Logout", destroy_admin_session_path, class: "mdi mdi-logout me-1  dropdown-item notify-item" %></span>
```
Comment (KhoaNDAHopee) [inline app/views/shared/partials/_navbar.html.erb]: Check indent.

File: config/database.yml [hunk 6]
```diff
@@ -5,18 +5,18 @@ default: &default
```
Comment (KhoaNDAHopee) [inline config/database.yml:5]: Don't push your local configs to github

File: spec/factories/admins.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+require 'faker'
+FactoryBot.define do
+  factory :admin do
+    first_name {Faker::Name.first_name}
```
Comment (KhoaNDAHopee) [inline spec/factories/admins.rb]: Check rubocop, should be ``{ Faker::Name.first_name }``

File: spec/factories/admins.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+require 'faker'
```
Comment (KhoaNDAHopee) [inline spec/factories/admins.rb]: No need require here.

**Review by KhoaNDAHopee — CHANGES_REQUESTED** [review]

File: app/views/devise/sessions/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,60 @@
+<html lang="en">
+<%= render partial: '/layouts/login/devise'%>
```
Comment (KhoaNDAHopee) [inline app/views/devise/sessions/new.html.erb]: Muốn sử dụng 1 layout khác (thay vì layout cho admin) cho devise sử dụng cách sau https://github.com/heartcombo/devise/wiki/How-To%3A-Create-custom-layouts

**Review by KhoaNDAHopee — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 10]
```diff
@@ -11,6 +11,8 @@
   namespace :administrator do
     # route for administrator
     root 'dashboard#index'
+        resources :admins
```
Comment (KhoaNDAHopee) [inline config/routes.rb]: Check indent

**Pull request merged**